packages/graphql-transformer-core/src/util/fileUtils.ts

```diff
@@ -0,0 +1,183 @@
+    if (!pathExists) {
+        return;
+    }
+    const dirStats = await lstat(directory);
+    if (!dirStats.isDirectory()) {
+        const buf = await readFile(directory);
+        return buf.toString();
+    }
+    const files = await readDir(directory);
+    const accum = {};
+    for (const fileName of files) {
+        const fullPath = path.join(directory, fileName);
+        const value = await readFromPath(fullPath);
+        accum[fileName] = value;
+    }
+    return accum;
+}
+
+/**
+ * Recursively clear the contents at some path. Might be a file or directory.
+ * @param clearPath The path to clear.
+ */
+export async function clearAtPath(clearPath: string) {
+    const pathExists = await exists(clearPath);
+    if (pathExists) {
+        const dirStats = await lstat(clearPath);
+        if (dirStats.isDirectory()) {
+            await deleteDirectory(clearPath);
+        } else {
+            await unlink(clearPath);
+        }
+    }
+}
+
+export type FileHandler = (file: { Key: string, Body: Buffer | string}) => Promise<string>;
+/**
+ * Uploads a file with exponential backoff up to a point.
+ * @param opts The deployment options
+ * @param key The bucket key
+ * @param body The blob body as a buffer
+ * @param backoffMS The time to wait this invocation
+ * @param numTries The max number of tries
+ */
+export async function handleFile(handler: FileHandler, key: string, body: Buffer, backoffMS: number = 500, numTries: number = 3) {
+    try {
+        return await handler({
+            Key: key,
+            Body: body
+        })
+    } catch (e) {
+        if (numTries > 0) {
+            await new Promise((res, rej) => setTimeout(() => res(), backoffMS))
+            await handleFile(handler, key, body, backoffMS * 2, numTries - 1)
+        }
+        throw e
+    }
+}
+
+export async function walkDirRec(
+    dir: string, handler: FileHandler, relativePath: string = '', joinPath: (...paths: string[]) => string
+) {
+    const files = await readDir(dir)
+    for (const file of files) {
+        const resourcePath = path.join(dir, file)
+        const newRelPath = joinPath(relativePath, file)
+        const isDirectory = (await lstat(resourcePath)).isDirectory()
+        if (isDirectory) {
+            await walkDirRec(resourcePath, handler, newRelPath, joinPath)
+        } else {
+            const resourceContents = await readFile(resourcePath);
+            await handleFile(handler, newRelPath, resourceContents)
+        }
+    }
+}
+
+export async function walkDir(dir: string, handler: (file: { Key: string, Body: Buffer | string}) => Promise<string>) {
+    return await walkDirRec(dir, handler, '', path.join);
+}
+export async function walkDirPosix(dir: string, handler: (file: { Key: string, Body: Buffer | string}) => Promise<string>) {
+    return await walkDirRec(dir, handler, '', path.posix.join);
+}
+
+export function throwIfNotJSONExt(stackFile: string) {
+    const nameParts = stackFile.split('.');
```

Comment:
we could use `path.extname` instead

Reply:
This package does not currently depend on fs-extra. I agree that we could move to use these but since we don't have the dependency right now I think I prefer the self contained changes for now. Thoughts?

Reply:
Updated.